src/cuda/multi_cuda/analysis.cu

```diff
@@ -0,0 +1,171 @@
+#include <iostream>
+#include <stdio.h>
+#include <vector>
+#include <array>
+#include "../cuda_common.cuh"
+
+#include <timing.h>
+#include "analysis.h"
+
+#define THREAD_COUNT 1024
+#define COLLISIONS_BEFORE_SWITCH (1)
+
+using namespace multi_cuda;
+
+__device__ inline Node& get_node(Node* nodes, int index, int node_size){
+    return *(Node*)(((int8_t*)nodes) + index * node_size);
+}
+
+__device__ void add_sucessors_to_worklist(int* successors, int work_columns[][THREAD_COUNT], int work_column_count, int current_work_column){
+    int initial_work_column = current_work_column;
+    for(int i = 0; i < 5; i++){
+        int succ_index = successors[i];
+        if (succ_index == -1)
+            return;
+        intptr_t hash = succ_index * 120811; 
+        int collision_count = 0;
+        int* work_column = work_columns[current_work_column];
+        while(work_column[hash % THREAD_COUNT] != -1){
+            if(work_column[hash % THREAD_COUNT] == succ_index){
+                break;
+            }
+            
+
+            if(++collision_count >= COLLISIONS_BEFORE_SWITCH){
+                current_work_column = (current_work_column + 1) % work_column_count;
+                work_column = work_columns[current_work_column];
+            }else{
+                hash++;
+            }
+        }
+        work_column[hash % THREAD_COUNT] = succ_index;
```

Comment:
Opportunity to reuse code from cuda-worklist?

Reply:
I was planning on getting it working first then moving into refactoring and finding some way to reuse the cuda code.